### PR TITLE
feat(D-1): implement ReviewToFixHandoff schema for Reviewer->Fixer interface (#3225)

### PR DESCRIPTION
**Pull request by RC918**
## Summary

Implements the stable interface contract between Reviewer and Coder Agent (EPIC D). This PR adds the `ReviewToFixHandoff` schema that defines how fix suggestions flow from Reviewer to Fixer.

**New components:**
- `FixSuggestion` model: Individual fix suggestion with file path, line range, code snippets, confidence score, and category
- `ReviewToFixHandoff` model: Container for suggestions with auto-fix eligibility signals and risk assessment
- `build_fix_handoff()`: Constructs handoff from suggestions and ReviewOutcome
- `should_route_to_fixer()`: Routing decision logic for Router
- `build_empty_handoff()`: Fallback for cases with no suggestions

**Key design decisions:**
- Schema version 1 with additive-only evolution strategy
- Auto-fix eligibility requires: low severity, no diff truncation, schema validated, high-confidence suggestions (>=0.8)
- `should_route_to_fixer()` trusts `auto_fix_eligible` flag and only adds hard safety check for max_severity
- `requires_human_review` is a parameter (default True), not computed - decision owned by Router/producer
- Security category infers high severity; bug_fix infers medium

## Updates since last revision

**Round 1 - Gemini Code Assist feedback:**

1. **Added `HIGH_CONFIDENCE_THRESHOLD` constant**: Extracted 0.8 threshold to module-level constant for maintainability. Exported from `__init__.py` for external use.

2. **Optimized `_is_auto_fix_eligible()`**: Replaced list comprehension with `any()` for efficiency - avoids allocating a list just to check existence.

3. **Fixed docstring example**: Updated Usage example to match actual function signatures (`should_route_to_fixer(handoff)` not `should_route_to_fixer(review_outcome, handoff)`).

**Round 2 - MorningAI Reviewer Agent feedback:**

4. **Added Security Warning for `requires_human_review`**: Added explicit warning in `build_fix_handoff()` docstring about the risks of setting `requires_human_review=False`. Lists 4 conditions that must be true before bypassing human review.

5. **Added boundary tests for confidence threshold**: New `TestConfidenceThresholdBoundary` class with 6 tests covering edge cases around the 0.8 threshold (exactly at, just below 0.79, just above 0.81, mixed confidence scenarios).

**Follow-up issues created:**
- #3234: Make HIGH_CONFIDENCE_THRESHOLD configurable via settings
- #3235: Add integration tests for ReviewToFixHandoff schema
- #3236: Document schema versioning and deprecation mechanism
- #3237: Align category-severity mapping with global taxonomy

## Review & Testing Checklist for Human

- [ ] **Security Warning clarity**: Review the new Security Warning in `build_fix_handoff()` docstring - is it clear enough to prevent misuse of `requires_human_review=False`?
- [ ] **Boundary test coverage**: Verify the 6 new boundary tests adequately cover edge cases around the 0.8 confidence threshold
- [ ] **API contract**: `should_route_to_fixer(handoff)` takes only one parameter. Verify no callers exist yet that would break.
- [ ] **Constant export**: `HIGH_CONFIDENCE_THRESHOLD` is now exported from `core.routing` - future consumers should use this constant.

**Test plan:** Run `pytest tests/test_fix_handoff.py -v` (41 tests). This is schema-only; no integration testing needed until a producer/consumer is implemented.

### Notes
- This is schema definition only - no existing code is modified to use this schema yet
- Follows same patterns as existing `ReviewOutcome` schema (B-6)
- Related: #3130 (B-6 Router Interface), EPIC D (Coder Agent Family)

Requested by: Ryan Chen (@RC918)
Link to Devin run: https://app.devin.ai/sessions/4c624f57c50e499f98859c62908c800d

### CODE REVIEW

File: handoff/20250928/40_App/orchestrator/core/routing/fix_handoff.py [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+    if not high_confidence:
+        logger.info("[SKIP_FIXER] no high-confidence suggestions")
+        return False
+
+    logger.info(
+        f"[ROUTE_TO_FIXER] pr_number={handoff.pr_number}, "
+        f"suggestions={len(handoff.suggestions)}, "
+        f"high_confidence={len(high_confidence)}"
+    )
+    return True
```
Comment (RC918) [inline handoff/20250928/40_App/orchestrator/core/routing/fix_handoff.py]: IMPACT: The `should_route_to_fixer` function repeats checks already performed in `_is_auto_fix_eligible` (like severity and confidence), creating redundant validation. This increases maintenance cost and risk of divergence. SUGGESTED FIX: Rely solely on `handoff.auto_fix_eligible` and `max_severity` in `should_route_to_fixer`, removing the duplicate high-confidence check since it's already part of auto-fix eligibility.

**Review by RC918 — COMMENTED** [review]
## MorningAI Code Review
